File: src/main/java/frc/team4069/robot/commands/vacuum/ToggleVacuumCommand.java
package frc.team4069.robot.commands.vacuum;

import edu.wpi.first.wpilibj.command.ConditionalCommand;
import frc.team4069.robot.subsystems.VacuumSubsystem;

// This can't use CommandBase, so I need to access the subsystem through its instance getter
// Deal with it.
public class ToggleVacuumCommand extends ConditionalCommand {

    public ToggleVacuumCommand() {
        super(new StopVacuumCommand(), new StartVacuumCommand());
        requires(VacuumSubsystem.getInstance());
    }

    @Override
    protected boolean condition() {
        return VacuumSubsystem.getInstance().isStarted();
    }
}

File: src/main/java/frc/team4069/robot/commands/vacuum/ToggleVacuumCommand.kt
package frc.team4069.robot.commands.vacuum

import edu.wpi.first.wpilibj.command.ConditionalCommand
import frc.team4069.robot.subsystems.VacuumSubsystem

class ToggleVacuumCommand : ConditionalCommand(StopVacuumCommand(), StartVacuumCommand()) {

    init {
        requires(VacuumSubsystem)
    }

    override fun condition(): Boolean = VacuumSubsystem.isStarted()

}